### PR TITLE
The Green-Tao theorem on arithmetic progressions in primes.

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

**Comment by spitters** [comment]
@andrejbauer This seems to give an interesting test case for your network theory.
Do we really want fabstracts to define `prime`? I guess it even exists in the lean std lib.

**Comment by andrejbauer** [comment]
I don't want to force contributors to search through all of fabstracts to find out where to link. I think we'll experience [preferential attachment](https://en.wikipedia.org/wiki/Preferential_attachment) and then it won't matter whether we try to build a system. It will build itself.

In this particular case I would ask the author whether he thinks that 1 is a prime. I think we have a rather fine example at hand in which a definition is wrong because it's not been used. It would be better to have an `unfinished prime`.

By the way, is Travis failing because of Travis or because of something in the PR? It seems that Travis is unhappy with the setup.

**Comment by robertylewis** [comment]
> Do we really want fabstracts to define prime? I guess it even exists in the lean std lib.

You might be surprised, for now.... But I'm sure it will exist at some point.

> By the way, is Travis failing because of Travis or because of something in the PR?

Travis is failing because #46 hasn't been merged (and this isn't based on top of that PR).

File: fabstract/Green_B_and_Tao_T_ArithmeticProgressionsInPrimes/fabstract.lean [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import meta_data
+
+namespace Green_B_and_Tao_T_ArithmeticProgressionsInPrimes
+
+def prime (n : nat) : Prop := ∀ m < n, (m = 0) ∨ (m = 1) ∨ (n % m ≠ 0)
```
Comment (andrejbauer) [inline fabstract/Green_B_and_Tao_T_ArithmeticProgressionsInPrimes/fabstract.lean]: By my reading this makes 1 a prime.

**Comment by spitters** [comment]
@andrejbauer well-spotted. That seems to be a reason to prefer more standard/popular definitions.
Re: use/testing,  Since there are no proofs in fabstracts, would some form of quickcheck be useful ?
We did it recently in an `easycrypt` project where one can use smt to proof "trivial" properties.
Perhaps lean's  automation is good enough already.

**Comment by kim-em** [comment]
I was wondering exactly the same thing --- how is fabstracts going to cope
with people defining things themselves over and over?

(Although I hadn't intended to make the issue so stark by giving the wrong
definition of a prime; oops!)

I would hope the process is something like:
* Contributions would be welcome, even with duplications and even with
errors.
* Ideally errors like my one here are noticed at the pull request stage and
fixed.
* If the contribution duplicates work already existing in the stdlib or in
the "common" parts of fabstracts, this can either be pointed out at the
pull request stage or just fixed by someone later.
* As foundational libraries (number theory, combinatorics, analysis,
topology, category theory, algebra, ...) are developed, the fabstracts
project just starts relying on them as leanpkg dependencies, as soon as
multiple fabstracts need the same material.
* If a single fabstract relies on a big and complicated library, we ...
cross that bridge when we come to it.

I understand there might be a worry about adding too many dependencies to
fabstracts, but we're far from that point now, and it's a good problem to
have.

**Review by andrejbauer — APPROVED** [review]

**Pull request merged**